packages/cloud_firestore/cloud_firestore/example/integration_test/web_snapshot_listeners.dart

```diff
@@ -0,0 +1,144 @@
+// Copyright 2020, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+// Run only on web for demonstrating snapshot listener clean up in debug mode does not clean up the listeners incorrectly.
+// See: https://github.com/firebase/flutterfire/issues/13019
+void runWebSnapshotListenersTests() {
+  group('Web snapshot listeners', () {
+    late FirebaseFirestore firestore;
+    late CollectionReference<Map<String, dynamic>> collection;
+    late DocumentReference<Map<String, dynamic>> document;
+    setUpAll(() async {
+      firestore = FirebaseFirestore.instance;
+      collection = firestore
+          .collection('flutter-tests/web-snapshot-listeners/query-tests');
+      document = collection.doc('doc1');
```

Comment:
Good it you can add a test with 2 ref on the same doc (since that was the original issue):
```
documentRef1 = collection.doc('doc1');
documentRef2 = collection.doc('doc1');
```

Reply:
done 👍